### PR TITLE
Replace level_mg_handler by mg_level

### PR DESCRIPTION
**Pull request by peterrum**
First (painless) suggestion.

FYI @kronbichler @tjhei @bangerth

### CODE REVIEW

File: include/deal.II/matrix_free/matrix_free.h [hunk 1]
```diff
@@ -166,7 +166,7 @@ class MatrixFree : public Subscriptor
    * local time stepping where one would like to control which elements
    * together form a batch of cells. The array `cell_vectorization_categories`
    * is accessed by the number given by cell->active_cell_index() when working
-   * on the active cells with `level_mg_handler` set to `-1` and by
+   * on the active cells with `mg_level` set to `-1` and by
```
Comment (tjhei) [inline include/deal.II/matrix_free/matrix_free.h]:
```suggestion
   * on the active cells with `mg_level` set to numbers::invalid_unsigned_int and by
```

File: include/deal.II/matrix_free/matrix_free.h [hunk 2]
```diff
@@ -381,7 +382,14 @@ class MatrixFree : public Subscriptor
...
      */
-    unsigned int level_mg_handler;
+    unsigned int mg_level;
+
+    /**
+     * Alias for mg_level
+     *
+     * @deprecated Use mg_level instead.
+     */
+    DEAL_II_DEPRECATED unsigned int &level_mg_handler;
```
Comment (tjhei) [inline include/deal.II/matrix_free/matrix_free.h]: excellent, that is a smart way to do it. :+1:

File: include/deal.II/matrix_free/matrix_free.h [hunk 3]
```diff
@@ -442,7 +450,7 @@ class MatrixFree : public Subscriptor
      * control which elements together form a batch of cells.
      *
      * This array is accessed by the number given by cell->active_cell_index()
-     * when working on the active cells with @p level_mg_handler set to -1 and
+     * when working on the active cells with @p mg_level set to -1 and
```
Comment (tjhei) [inline include/deal.II/matrix_free/matrix_free.h]:
```suggestion
     * when working on the active cells with @p mg_level set to numbers::invalid_unsigned_int and
```

File: include/deal.II/matrix_free/matrix_free.h [hunk 4]
```diff
@@ -1605,7 +1613,7 @@ class MatrixFree : public Subscriptor
    * Return the level of the mesh to be worked on.
    */
   unsigned int
-  get_level_mg_handler() const;
+  get_mg_level() const;
```
Comment (tjhei) [inline include/deal.II/matrix_free/matrix_free.h]: Can you add "Returns numbers::invalid_unsigned_int if working on active cells."

**Comment by peterrum** [comment]
@bangerth I guess we are waiting for your approval of the name ;)

File: include/deal.II/matrix_free/matrix_free.h (hunk 2 above)
Comment (kronbichler) [reply include/deal.II/matrix_free/matrix_free.h]: Smart yes, but the reference variable did remove the implicit copy constructor of `MatrixFree::AdditionalData` that was previously available. I think we should introduce an explicit copy operator to restore backward compatibility.

Comment (peterrum) [reply include/deal.II/matrix_free/matrix_free.h]: I'll do!

**Pull request merged**